internal/service/provisioning/api/openapi.yaml

```diff
@@ -0,0 +1,290 @@
+                $ref: "../../common/api/openapi.yaml#/components/schemas/ProblemDetails"
+        '412':
+          description: Precondition failed.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../../common/api/openapi.yaml#/components/schemas/ProblemDetails"
+        '500':
+          description: Internal server error.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../../common/api/openapi.yaml#/components/schemas/ProblemDetails"
+    delete:
+      operationId: deleteProvisioningRequest
+      summary: Delete a provisioning request
+      description: |
+        Deletes a provisioning request.
+      parameters:
+      - $ref: "#/components/parameters/provisioningRequestId"
+      tags:
+      - provisioningRequests
+      responses:
+        '200':
+          description: Successfully deleted the provisioning request.
+        '404':
+          description: The specified provisioning request was not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../../common/api/openapi.yaml#/components/schemas/ProblemDetails"
+        '500':
+          description: Internal server error.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "../../common/api/openapi.yaml#/components/schemas/ProblemDetails"
+
+components:
+  parameters:
+    provisioningRequestId:
+      name: provisioningRequestId
+      description: |
+        Unique identifier of a provisioning request.
+      in: path
+      required: true
+      schema:
+        type: string
+        format: uuid
+      example: 123e4567-e89b-12d3-a456-426614174000
+
+  schemas:
+    ProvisioningRequest:
+      description: |
+        Information about a provisioning request.
+      type: object
+      properties:
+        provisioningRequestId:
+          type: string
+          format: uuid
+          description: |
+            Identifier for the provisioning request. This identifier is assigned by the SMO.
+          example: "123e4567-e89b-12d3-a456-426614174000"
+        name:
+          type: string
+          description: Human readable name of the provisioning request.
+        description:
+          type: string
+          description: Human readable description of the provisioning request.
+        templateName:
+          type: string
+          description: Name of the template used for the provisioning request.
+        templateVersion:
+          type: string
+          description: Version of the template used for the provisioning request.
+        templateParameters:
+          type: object
+          description: Input Parameters that conform to the OpenAPI V3 schema defined in the template.
+        nodeClusterId:
+          type: string
+          format: uuid
+          description: Identifier of the NodeCluster that has been provisioned.
```

Comment:
Is `nodeClusterId` part of the request or is this a status type field that is only populated once we've create the `NodeCluster`?  If it is status, then would it be more appropriate to move this to the `status` section?

cc: @bartwensley @browsell

Reply:
Part of the status, should move to the status section.

Reply:
Thanks for pointing this out. Will update.